turtle_graphics/turtle_graphics.js

```diff
@@ -0,0 +1,310 @@
+                }
+                //console.log("Current length of array: " + this.allPositions.length);
+                //console.log("Final position after calling Forward4: " + "[X: " + this.x + ", Y: " + this.y + "]");
+            }
+        } else if (num < 0) {
+            //do nothing, i.e. keep the position and direction of the turtle as is
+            //console.log("Current direction of turtle is: " + this.direction);
+            console.log("A negative value was entered into the forward method. Non complaint. ")
+            console.log("Last position of turtle is at (x,y): " + this.allPositions[this.allPositions.length - 1]);
+            console.log("Only coordinates up to this point will be printed. Please restart");
+        } else {
+            console.log("Exception case: not handled yet");
+        }
+
+        return this;
+    }
+
+    right() {
+
+        //console.log('Current direction BEFORE turning RIGHT: ' + this.direction);
+        if (this.direction === 'EAST') {
+            this.direction = 'SOUTH';
+        } else if (this.direction === 'WEST') {
+            this.direction = 'NORTH';
+        } else if (this.direction === 'NORTH') {
+            this.direction = 'EAST';
+        } else if (this.direction === 'SOUTH') {
+            this.direction = 'WEST';
+        }
+        return this;
+        //console.log('Current direction AFTER turning RIGHT: ' + this.direction);
+    }
+
+    left() {
+
+        //console.log('Current direction BEFORE turning LEFT: ' + this.direction);
+        if (this.direction === 'EAST') {
+            this.direction = 'NORTH';
+        } else if (this.direction === 'WEST') {
+            this.direction = 'SOUTH';
+        } else if (this.direction === 'NORTH') {
+            this.direction = 'WEST';
+        } else if (this.direction === 'SOUTH') {
+            this.direction = 'EAST';
+        }
+        return this;
+        //console.log('Current direction AFTER turning LEFT: ' + this.direction);
+
+    }
+
+    allPoints() {
+
+        //simply returns already existing allPositions array that belongs to the turtle class
+        
+        //let length = this.allPositions.length;
+        //console.log("Current allPosition length: " + length);
+        //if (length >= 0) { //i.e. the turtle has moved at all
+        //for (let i = 0; i < length; i++) {
+          // console.log(this.allPositions[i]);
+        //}
+        
+        return this.allPositions;
+
+    }
+
+    print() {
+
+        let allPoints = this.allPositions;
+
+        let maxX = Math.max(...allPoints.map(element => {
+            return element[0];
+        })); //columns
+
+        //console.log("Max x: " + maxX);
+
+        let maxY = Math.max(...allPoints.map(element => {
+            return element[1];
+        }));//rows
+
+        //console.log("Max y: " + maxY);
+
+        let a1 = []; //temporarily array of array to hold all x,y grid coordinates
```

Comment:
Instead of a comment, can name the variable something like

`gridCoordinations = []`